library/rsa.c

```diff
@@ -1550,7 +1550,12 @@ int mbedtls_rsa_rsassa_pss_sign( mbedtls_rsa_context *ctx,
         return( MBEDTLS_ERR_RSA_BAD_INPUT_DATA );
 
     hlen = mbedtls_md_get_size( md_info );
-    slen = hlen;
+
+    /* Align with FIPS-186-4 */
+    if ( olen == 128 && hlen == 64 )
+        slen = hlen - 2;
+    else
+        slen = hlen;
 
     if( olen < hlen + slen + 2 )
```

Comment:
I think this should move to the `else` statement before, in a block, as in case `olen` is `128` and `hlen` is 64, this is a redundant check.  ( 128 is always equal to 64 + 64 - 2 + 2 )

Reply:
It is related to the encoding: `zeros || hash length || salt` please see step 5 in the following:
https://tools.ietf.org/html/rfc8017#page-43

Reply:
I know, but what I mean is change this to:
~~~
    if ( olen == 128 && hlen == 64 )
    {
         slen = hlen - 2;
    }
    else
    {
        slen = hlen;
        if( olen < hlen + slen + 2 )
            return( MBEDTLS_ERR_RSA_BAD_INPUT_DATA );
    }
This is because:
a.  It is redundant to check that 128 < 128 . Although it may be possible that the compiler will optimize this out.
b. The link you referenced is an IETF RFC, which is relevant to the `else` statement, as the use case of `olen` 128 and `hlen` 64 is FIPS 186-4 related.

Reply:
I don't think it's a must, but I think it's more readable, and no redundant check

Reply:
Done